apm-agent-plugins/apm-grpc/apm-grpc-plugin/src/test/java/co/elastic/apm/agent/grpc/testapp/HelloClient.java

```diff
@@ -37,17 +38,23 @@
 import java.util.concurrent.Future;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicLong;
 import java.util.concurrent.atomic.AtomicReference;
 import java.util.stream.Collectors;
 
 public abstract class HelloClient<Req, Rep> {
 
     private static final Logger logger = LoggerFactory.getLogger(HelloClient.class);
-
     private final ManagedChannel channel;
+    private final AtomicLong errorCount;
 
     protected HelloClient(ManagedChannel channel) {
         this.channel = channel;
+        this.errorCount = new AtomicLong(0);
+    }
+
+    protected static Deadline getDeadline() {
```

Comment:
adding a deadline makes remote calls fail with timeout instead of waiting forever.